src/components/AccentPhrase.vue

```diff
@@ -214,95 +227,55 @@ const handleChangePronounce = (newPronunciation: string) => {
 };
 
-type hoveredType = "vowel" | "consonant";
-
-type hoveredInfoType = {
-  accentPhraseIndex: number | undefined;
-  moraIndex?: number | undefined;
-  type?: hoveredType;
-};
-
-const accentHoveredInfo = reactive<hoveredInfoType>({
-  accentPhraseIndex: undefined,
-});
-
-const pitchHoveredInfo = reactive<hoveredInfoType>({
-  accentPhraseIndex: undefined,
-  moraIndex: undefined,
-});
-
-const lengthHoveredInfo = reactive<hoveredInfoType>({
-  accentPhraseIndex: undefined,
-  moraIndex: undefined,
-  type: "vowel",
-});
-
+const hoveredMoraIndex = ref<number | undefined>(undefined);
 const handleHoverText = (isOver: boolean, moraIndex: number) => {
-  if (props.selectedDetail == "accent") {
-    if (isOver) {
-      accentHoveredInfo.accentPhraseIndex = props.index;
-    } else {
-      accentHoveredInfo.accentPhraseIndex = undefined;
-    }
-  } else if (props.selectedDetail == "pitch") {
-    if (isOver) {
-      pitchHoveredInfo.accentPhraseIndex = props.index;
-      pitchHoveredInfo.moraIndex = moraIndex;
-    } else {
-      pitchHoveredInfo.accentPhraseIndex = undefined;
-      pitchHoveredInfo.moraIndex = undefined;
-    }
+  if (props.selectedDetail == "accent" || props.selectedDetail == "pitch") {
+    hoveredMoraIndex.value = isOver ? moraIndex : undefined;
   }
 };
 
+const lengthHoveredPhonemeType = ref<"vowel" | "consonant">("vowel");
 const handleLengthHoverText = (
   isOver: boolean,
   phoneme: MoraDataType,
-  phraseIndex: number,
-  moraIndex?: number
+  moraIndex: number
 ) => {
   if (phoneme !== "vowel" && phoneme !== "consonant")
     throw new Error("phoneme != hoveredType");
-  lengthHoveredInfo.type = phoneme;
+  lengthHoveredPhonemeType.value = phoneme;
   // the pause and pitch templates don't emit a mouseOver event
-  if (isOver) {
-    lengthHoveredInfo.accentPhraseIndex = phraseIndex;
-    lengthHoveredInfo.moraIndex = moraIndex;
-  } else {
-    lengthHoveredInfo.accentPhraseIndex = undefined;
-    lengthHoveredInfo.moraIndex = undefined;
-  }
+  hoveredMoraIndex.value = isOver ? moraIndex : undefined;
 };
 
 const unvoicableVowels = ["U", "I", "i", "u"];
 
-const isHovered = (vowel: string, moraIndex: number) => {
-  let isHover = false;
-  if (!uiLocked.value) {
-    if (props.selectedDetail == "accent") {
-      if (props.index === accentHoveredInfo.accentPhraseIndex) {
-        isHover = true;
-      }
-    } else if (props.selectedDetail == "pitch") {
-      if (
-        props.index === pitchHoveredInfo.accentPhraseIndex &&
-        moraIndex === pitchHoveredInfo.moraIndex &&
-        unvoicableVowels.includes(vowel)
-      ) {
-        isHover = true;
-      }
-    }
+/**
+ * 各モーラが、hover中のモーラをそのままクリックした場合に編集範囲に含まれるかどうか。
+ * 強調表示するかの判定に使われる。
+ */
+const isEditableMora = (vowel: string, moraIndex: number) => {
+  if (uiLocked.value) {
+    return false;
   }
-  return isHover;
+  if (props.selectedDetail == "accent") {
+    // クリック時の動作はアクセント句全体の読み変更。
+    // よって、いずれかのモーラがhoverされているならアクセント句全体を強調表示する。
```

Comment:
`アクセント句全体`だと全アクセント句のことだと誤解されるかもなので、`そのアクセント句`とかがよりあってるかも。
そこだけ変えてマージさせていただきます！
```suggestion
    // クリック時の動作はそのアクセント句の読み変更。
    // よって、いずれかのモーラがhoverされているならそのアクセント句を強調表示する。
```